src/type_checkers/negated_expression.cr

```diff
@@ -0,0 +1,7 @@
+module Mint
+  class TypeChecker
+    def check(node : Ast::NegatedExpression) : Type
+      BOOL
```

Comment:
I'm wondering if we need to constraint it's expression to BOOL, since there is no `null` or `undefined` what other use cases are there that needs converting things to a boolean.

Reply:
I... just don't trust JS, but I'm also not entirely sure what this TypeChecker does.

Reply:
The type checker allows to compare two types see if they are (or can be the same). A good example is the operation, for the `-` `*` and `/` operations both of the operands must be a number https://github.com/mint-lang/mint/blob/master/src/type_checkers/operation.cr#L45 

What is returned here will be the type of the negated expression, in this case always a boolean.

We could say that only booleans can be negated by evaluating the type of the expression, and raising an error if it's not a boolean (BOOL).